sshd-sftp/src/main/java/org/apache/sshd/server/subsystem/sftp/AbstractSftpSubsystemHelper.java

```diff
@@ -1631,8 +1650,8 @@ protected void doRemove(int id, String path, LinkOption... options) throws IOExc
             log.debug("signalRemovalPreConditionFailure(id={})[{}] signal {} for {}: {}",
                 id, pathValue, thrown.getClass().getSimpleName(), path, thrown.getMessage());
         }
-        listener.removing(session, path);
-        listener.removed(session, path, thrown);
+        listener.removingFile(session, path);
+        listener.removedFile(session, path, thrown);
```

Comment:
Not sure this is correct - {{signalRemovalPreConditionFailure}} may be called for both file and directory so by what right do we decide here that it is a file and not a directory ?

Reply:
Thats right. It is being called for both directory and file. Fixed the same.